### ISSUE TITLE
Fix clang after rebase on branch release/16.x

### ISSUE THREAD

**Issue by aleasims** [NilFoundation/zkllvm-circifier#27]
Failed to compile `clang` on branch `release/16.x`:

```
/root/proj/zkllvm/libs/circifier/clang/lib/Basic/Targets/Assigner.cpp:25:31: error: could not convert ‘clang::ALL_LANGUAGES’ from ‘clang::LanguageID’ to ‘clang::HeaderDesc’
   25 |   {#ID, TYPE, ATTRS, nullptr, ALL_LANGUAGES, nullptr},
      |                               ^~~~~~~~~~~~~
      |                               |
      |                               clang::LanguageID
/root/proj/zkllvm/libs/circifier/clang/include/clang/Basic/BuiltinsAssigner.def:21:1: note: in expansion of macro ‘BUILTIN’
   21 | BUILTIN(__builtin_assigner_malloc, "v*z", "n")
      | ^~~~~~~
```

Also got a deprecation warning:
```
/root/proj/zkllvm/libs/circifier/clang/lib/Basic/Targets/Assigner.cpp: In member function ‘virtual llvm::ArrayRef<clang::Builtin::Info> clang::targets::AssignerTargetInfo::getTargetBuiltins() const’:
/root/proj/zkllvm/libs/circifier/clang/lib/Basic/Targets/Assigner.cpp:43:30: warning: ‘llvm::ArrayRef<T> llvm::makeArrayRef(const T*, size_t) [with T = clang::Builtin::Info; size_t = long unsigned int]’ is deprecated: Use deduction guide instead [-Wdeprecated-declarations]
   43 |     return llvm::makeArrayRef(BuiltinInfo, clang::assigner::LastTSBuiltin -
      |            ~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   44 |                                              Builtin::FirstTSBuiltin);
      |                                              ~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /root/proj/zkllvm/libs/circifier/llvm/include/llvm/Support/HashBuilder.h:18,
                 from /root/proj/zkllvm/libs/circifier/clang/include/clang/Basic/Sanitizers.h:19,
                 from /root/proj/zkllvm/libs/circifier/clang/include/clang/Basic/CodeGenOptions.h:17,
                 from /root/proj/zkllvm/libs/circifier/clang/include/clang/Basic/TargetInfo.h:19,
                 from /root/proj/zkllvm/libs/circifier/clang/lib/Basic/Targets/Assigner.h:16,
                 from /root/proj/zkllvm/libs/circifier/clang/lib/Basic/Targets/Assigner.cpp:13:
/root/proj/zkllvm/libs/circifier/llvm/include/llvm/ADT/ArrayRef.h:517:15: note: declared here
  517 |   ArrayRef<T> makeArrayRef(const T *data, size_t length) {
      |               ^~~~~~~~~~~~
```

**Comment by aleasims** [comment]
Compilation error fixed in 58b3b62fd7cf9ef5ce4df6a22adf17839bf8d4c9

**Comment by aleasims** [comment]
Warning fixed in da216b6bf2c9ea2e61c0fcdae380f56ab338e949
Closing
